x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/forms/rule_actions_form.tsx

```diff
@@ -0,0 +1,216 @@
+  onClose: () => void;
+  onConfirm: (bulkActionEditPayload: BulkActionEditPayload) => void;
+}
+
+const RuleActionsFormComponent = ({ rulesCount, onClose, onConfirm }: RuleActionsFormProps) => {
+  const {
+    services: {
+      triggersActionsUi: { actionTypeRegistry },
+    },
+  } = useKibana();
+
+  const formSchema = useMemo(() => getFormSchema(actionTypeRegistry), [actionTypeRegistry]);
+
+  const { form } = useForm({
+    schema: formSchema,
+    defaultValue: defaultFormData,
+  });
+
+  const [{ overwrite, throttle }] = useFormData({ form, watch: ['overwrite', 'throttle'] });
+
+  const handleSubmit = useCallback(async () => {
+    const { data, isValid } = await form.submit();
+    if (!isValid) {
+      return;
+    }
+
+    const { actions = [], throttle: throttleToSubmit, overwrite: overwriteValue } = data;
+    const editAction = overwriteValue
+      ? BulkActionEditType.set_rule_actions
+      : BulkActionEditType.add_rule_actions;
+
+    onConfirm({
+      type: editAction,
+      value: {
+        actions: actions.map(({ actionTypeId, ...action }) => action),
+        throttle: throttleToSubmit,
+      },
+    });
+  }, [form, onConfirm]);
+
+  const throttleFieldComponentProps = useMemo(
+    () => ({
+      idAria: 'bulkEditRulesRuleActionThrottle',
+      dataTestSubj: 'bulkEditRulesRuleActionThrottle',
+      hasNoInitialSelection: false,
+      euiFieldProps: {
+        options: THROTTLE_OPTIONS,
+      },
+    }),
+    []
+  );
+
+  const messageVariables = useMemo(() => getAllActionMessageParams(), []);
+
+  const showActionsSelect = throttle !== NOTIFICATION_THROTTLE_NO_ACTIONS;
+
+  return (
+    <BulkEditFormWrapper
+      form={form}
+      title={i18n.FORM_TITLE}
+      onClose={onClose}
+      onSubmit={handleSubmit}
+      flyoutSize="l"
+    >
+      <EuiCallOut
+        color="primary"
+        data-test-subj="bulkEditRulesRuleActionInfo"
+        title={
+          <FormattedMessage
+            id="xpack.securitySolution.detectionEngine.rules.allRules.bulkActions.edit.addRuleActions.infoCalloutTitle"
+            defaultMessage="Add actions for {rulesCount, plural, one {# rule} other {# rules}} you’ve selected. Here some tips and suggestions:"
+            values={{ rulesCount }}
+          />
+        }
+      >
+        <ul>
+          <li>
+            <FormattedMessage
+              id="xpack.securitySolution.detectionEngine.rules.allRules.bulkActions.edit.addRuleActions.actionFrequencyDetail"
+              defaultMessage="Action frequency is shared to all new and existing actions."
+            />
+          </li>
+          <li>
+            <FormattedMessage
+              id="xpack.securitySolution.detectionEngine.rules.allRules.bulkActions.edit.addRuleActions.deleteActionsDetail"
+              defaultMessage='If you’d like to delete actions for all selected rules, select "Perform no actions" in the dropdown and check Overwrite all selected rule actions.'
+            />
+          </li>
```

Comment:
Using `FormattedMessage` is something that expected in most of the cases https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/GUIDELINE.md#usage-of-appropriate-component

Had `RULE_VARIABLES_DETAIL` in translations as FormattedMessage component for some reason didn't work with encoded curly braces symbols. So I had to resort to using of `i18n.translate`

Reply:
@vitaliidm FWIW I'd expect `i18n.translate()` to be used most of the time, because usually what we need is a simple localized string. When we need something more, like a custom localized markup, then `FormattedMessage` might help as it supports some customizations like a `children` function etc.

The guide says:

> In the case where the string is expected (aria-label, placeholder), Call JS function i18n.translate() from the@kbn/i18n package.

In other words, by default we extract translations into a separate file and use this syntax:

```ts
  THROTTLE_HELP_TEXT: i18n.translate(
    'xpack.securitySolution.detectionEngine.rules.allRules.bulkActions.edit.addRuleActions.throttleHelpText',
    {
      defaultMessage:
        'Select when automated actions should be performed if a rule evaluates as true.',
    }
  ),
```

Reply:
> I think we would need to align on approach here within a team

Sounds good, I added this to our next Tech agenda 👍

Reply:
Alright, so everyone thinks that `i18n.translate` should be used by default.